app/src/main/java/com/duckduckgo/app/cta/onboarding_experiment/DaxDialogExperiment.kt

```diff
@@ -0,0 +1,225 @@
+
+    private var daxText: String = ""
+    private var primaryButtonText: String = ""
+    private var daxDialogListener: DaxDialogExperimentListener? = null
+    private var trackers: List<Entity> = listOf()
+
+    override fun setDaxDialogListener(listener: DaxDialogListener?) {
+        daxDialogListener = listener as DaxDialogExperimentListener?
+    }
+
+    fun setBlockedTrackers(blockedTrackers: List<Entity>) {
+        trackers = blockedTrackers
+    }
+
+    override fun onAttach(context: Context) {
+        AndroidSupportInjection.inject(this)
+        super.onAttach(context)
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val dialog = super.onCreateDialog(savedInstanceState)
+        dialog.window?.apply {
+            setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
+            statusBarColor = Color.TRANSPARENT
+        }
+        return dialog
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.run {
+            if (containsKey(ARG_DAX_TEXT)) {
+                getString(ARG_DAX_TEXT)?.let { daxText = it }
+            }
+            if (containsKey(ARG_PRIMARY_CTA_TEXT)) {
+                getString(ARG_PRIMARY_CTA_TEXT)?.let { primaryButtonText = it }
+            }
+        }
+    }
+
+    override fun getTheme(): Int {
+        return com.duckduckgo.mobile.android.R.style.Widget_DuckDuckGo_DaxDialogFragment
+    }
+
+    override fun onViewCreated(
+        view: View,
+        savedInstanceState: Bundle?,
+    ) {
+        super.onViewCreated(view, savedInstanceState)
+        setStepOneView()
+    }
+
+    override fun onDismiss(dialog: DialogInterface) {
+        if (activity != null) {
+            binding.dialogText.cancelAnimation()
+            daxDialogListener?.onDaxDialogDismiss()
+        }
+        daxDialogListener = null
+        super.onDismiss(dialog)
+    }
+
+    private fun setStepOneView() {
+        setDialog()
+        setListeners()
+        setLottieViewAnimation(SHOW_TRACKERS, this::setExpandedTrackersAndShowDialog)
+    }
+
+    private fun setExpandedTrackersAndShowDialog() {
+        setLottieViewAnimation(SHOW_TRACKERS_EXPANDED, this::setHandLoop)
+        binding.cardView.animate().alpha(1.0f).duration = 1000
+        binding.logo.animate().alpha(1.0f).duration = 1000
+        binding.dialogText.startTypingAnimation(daxText, true)
+    }
+
+    private fun setDialog() {
+        context?.let {
+            with(binding) {
+                cardView.alpha = 0.0f
+                logo.alpha = 0.0f
+                hiddenText.text = daxText.html(it)
+                primaryCta.text = primaryButtonText
+                dialogText.typingDelayInMs = DEFAULT_TYPING_DELAY
+            }
+        } ?: dismiss()
+    }
+
+    private fun setListeners() {
+        with(binding) {
+            primaryCta.setOnClickListener {
+                dialogText.cancelAnimation()
```

Comment:
We are missing a `daxDialogListener?.onDaxDialogPrimaryCtaClick()` here to also fire the Ok button pixel.